ipns/record.go

```diff
@@ -89,10 +93,29 @@ func (rec *Record) Value() (path.Path, error) {
 		return nil, err
 	}
 
-	p, err := path.NewPath(string(value))
+	if len(value) == 0 {
+		// To maximize interop across implementations, turn empty value into zero-length identity CID.
+		// This is a convenience placeholder used when the value in record is empty or does
+		// not matter because IPNS is used for custom CBOR in the Data field.
+		value = []byte(NoopValue)
+	}
+
+	// parse as a string with content path
+	if len(value) > 0 && value[0] == '/' {
+		p, err := path.NewPath(string(value))
+		if err != nil {
+			return nil, multierr.Combine(ErrInvalidPath, err)
+		}
+		// done, finish fast
+		return p, nil
+	}
+
+	// fallback: for legacy/optimization reasons, the value could be a valid CID in byte form
+	binaryCid, err := cid.Cast(value)
 	if err != nil {
 		return nil, multierr.Combine(ErrInvalidPath, err)
```

Comment:
Maybe use `path.ErrInvalidPath`?

Reply:
this is on purpose, to have IPNS type that works with `errors.Is` and `errors.As` but also human-readable information that error occurred in CID fallback

debugging IPNS is hard, this makes it bit easier